deliverables/ruby_monk_problems/restaurant.rb

```diff
@@ -4,6 +4,10 @@ def initialize(menu)
   end
 
   def cost(*orders)
-    # your code here
+    orders.inject(0) do |total_cost, order|
+      total_cost + order.keys.inject(0) {|cost, key| cost + @menu[key]*order[key] }
+    end
   end
 end
+#where did "key" come from?
```

Comment:
`key` is defined at the beginning of the block, inside `|cost, key|`.  

The inject method iterates over an array (in this case `order.keys`).  The first time through `cost` is set to `0`; each time after that cost is the return value from the previous time running the block.  So basically `cost` here could be named `running_total`